lib/oracle.js

```diff
@@ -350,7 +350,13 @@ Oracle.prototype.create = function (model, data, callback) {
   var params = fields.params;
   params.push(outParam);
   this.query(sql, params, function (err, info) {
-    if (err) return callback(err);
+    if (err) {
+      if (err.toString().indexOf('ORA-00001: unique constraint') >= 0) {
+        // Transform the error so that duplicate can be checked using regex
+        err = new Error(err.toString() + '. Duplicate id detected.');
```

Comment:
The error message is misleading unless "ORA-0001: unique constraint" is reported only for primary keys. I think the following text would server better: "Duplicate entry detected."